noir_stdlib/src/uint128.nr

```diff
@@ -91,23 +99,36 @@ impl U128 {
         if ascii < 58 {
             ascii - 48
         } else if ascii < 71 {
+            assert(ascii >= 65); // enforce >= 'A'
             ascii - 55
         } else {
+            assert(ascii >= 97); // enforce >= 'a'
+            assert(ascii <= 102); // enforce <= 'f'
             ascii - 87
         } as Field
     }
 
+    // TODO: Replace with a faster version. 
+    // A circuit that uses this function a lot will be really slow to compute
```

Comment:
The function can be replaced with a faster one, indeed, but we are not doing binary search and it is not 'really slow'.